File: problems/search_a_2d_matrix_ii/solution.py
class Solution:
    def searchMatrix(self, matrix, target):
        """
        :type matrix: List[List[int]]
        :type target: int
        :rtype: bool
        """
        if not matrix: return False
        y, x = len(matrix)-1, 0
        while y >= 0 and x < len(matrix[0]):
            if target == matrix[y][x]: return True
            if target < matrix[y][x]: y -= 1
            else: x += 1
        return False

File: problems/search_a_2d_matrix_ii/solution.cpp
class Solution {
public:
    bool searchMatrix(vector<vector<int>>& matrix, int target) {
        int y = matrix.size()-1, x = 0;
        while (y >= 0 && x < matrix[0].size()) {
            int val = matrix[y][x];
            if (val == target) return true;
            if (val > target) y--;
            if (val < target) x++;
        }
        return false;
    }
};